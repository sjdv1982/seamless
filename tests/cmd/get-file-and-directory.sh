#!/bin/bash
set -u
ASSISTANT=$1
rm -rf dummy-directory dummy-bufferfolder
mkdir dummy-directory dummy-bufferfolder
export SEAMLESS_READ_BUFFER_FOLDERS=$(pwd)/dummy-bufferfolder
seamless-delegate-stop >& /dev/stderr
seamless-delegate $ASSISTANT >& /dev/stderr

echo Text > dummy-directory/text.txt
mkdir dummy-directory/sub
echo 1 > dummy-directory/sub/one.txt
echo 2 > dummy-directory/sub/two.txt
seamless-upload -v --dest dummy-bufferfolder dummy-directory/text.txt >& /dev/stdout
seamless-upload -v --dest dummy-bufferfolder dummy-directory/sub >& /dev/stdout
echo Repeat, no files will be uploaded...
seamless-upload -v --dest dummy-bufferfolder dummy-directory/text.txt >& /dev/stdout
seamless-upload -v --dest dummy-bufferfolder dummy-directory/sub >& /dev/stdout
seamless-bufferdir-deploy-deepfolder $(cat dummy-directory/sub.CHECKSUM) dummy-bufferfolder
rm -rf dummy-directory/sub
rm -rf dummy-directory/text.txt
seamless 'cat dummy-directory/text.txt' >& /dev/stdout
seamless 'cd dummy-directory/sub; head *' >& /dev/stdout
rm -rf dummy-directory dummy-bufferfolder
