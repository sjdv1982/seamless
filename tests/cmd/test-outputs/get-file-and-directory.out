seamless: Verbosity set to 1
seamless: Upload 1 files, total 5.00 B
seamless: Verbosity set to 1
seamless: Upload 3 files, total 169.00 B
Repeat, no files will be uploaded...
seamless: Verbosity set to 1
seamless: Upload no files
seamless: Verbosity set to 1
seamless: Upload no files
Text

==> one.txt <==
1

==> two.txt <==
2

